osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -243,5 +235,32 @@ public DaySeparator(DateTimeOffset time)
                 };
             }
         }
+
+        /// <summary>
+        /// An <see cref="OsuScrollContainer"/> with functionality to automatically scrolls whenever the maximum scrollable distance increases.
+        /// </summary>
+        private class ChannelScrollContainer : OsuScrollContainer
+        {
+            private const float auto_scroll_leniency = 10f;
+
+            private float? lastExtent;
+
+            /// <summary>
+            /// Whether this should automatically scroll to end on the next call to <see cref="UpdateAfterChildren"/>.
+            /// </summary>
+            public bool ShouldAutoScroll { get; private set; } = true;
+
+            protected override void UpdateAfterChildren()
+            {
+                base.UpdateAfterChildren();
+
+                if ((lastExtent == null || ScrollableExtent > lastExtent) && ShouldAutoScroll)
+                    ScrollToEnd();
+                else
+                    ShouldAutoScroll = IsScrolledToEnd(auto_scroll_leniency);
```

Comment:
i don't know about this. feels flaky that this is in the `else` conditional. just trying to work my head around the flow of logic here is...

Reply:
can move this out of the `else` conditional, I agree it doesn't read well this way, it's just that the code will be an unnecessary set if the above branch is executed.

Reply:
Would agree with unconditionally setting this. It's a few floating point operations. It really shouldn't tank performance in any discernible manner.

In fact, because of the scheduled-after-children `ScrollToEnd()` earlier in this file, that doesn't update this flag's value, I'd say it's more correct to always set it. (After-children scheduler runs before `UpdateAfterChildren()`, I checked.)